src/runtime-rs/crates/hypervisor/src/remote/inner.rs

```diff
@@ -66,7 +70,47 @@ impl RemoteInner {
         Ok(HypervisorClient::new(c))
     }
 
-    pub(crate) async fn prepare_vm(&mut self, id: &str, netns: Option<String>) -> Result<()> {
+    fn prepare_annotations(
+        &self,
+        oci_annotations: &HashMap<String, String>,
+    ) -> HashMap<String, String> {
+        let mut annotations: HashMap<String, String> = HashMap::new();
+        let config = &self.config;
+        annotations.insert(
+            SANDBOX_NAME_LABEL_KEY.to_string(),
+            oci_annotations
+                .get(SANDBOX_NAME_LABEL_KEY)
+                .cloned()
+                .unwrap_or_default(),
+        );
+        annotations.insert(
+            SANDBOX_NAMESPACE_LABEL_KEY.to_string(),
+            oci_annotations
+                .get(SANDBOX_NAMESPACE_LABEL_KEY)
+                .cloned()
+                .unwrap_or_default(),
+        );
+        annotations.insert(
+            KATA_ANNO_CFG_HYPERVISOR_MACHINE_TYPE.to_string(),
+            config.machine_info.machine_type.to_string(),
+        );
+        annotations.insert(
+            KATA_ANNO_CFG_HYPERVISOR_DEFAULT_VCPUS.to_string(),
+            "0".to_string(),
+        );
+        annotations.insert(
+            KATA_ANNO_CFG_HYPERVISOR_DEFAULT_MEMORY.to_string(),
+            "0".to_string(),
+        );
```

Comment:
Why don't we use hypervisor config to supply values for vcpus and memory?

Also, how about image path?

Reply:
Thanks for your carefulness, I noticed that patch of enable image path annotation for remote hypervisor already merged 2 month ago, I will add it soon.

Reply:
If you feel though that this is out of scope for this PR (which has "basic boilerplate" in its title after all, not "full implementation" :-)) just let me know.